src/app/commands/cmd_symmetry_mode.cpp

```diff
@@ -46,7 +76,14 @@ bool SymmetryModeCommand::onChecked(Context* ctx)
 void SymmetryModeCommand::onExecute(Context* ctx)
 {
   auto& enabled = Preferences::instance().symmetryMode.enabled;
-  enabled(!enabled());
+  DocumentPreferences& docPref = Preferences::instance().document(ctx->activeDocument());
+  docPref.symmetry.mode.setValue(app::gen::SymmetryMode(this->m_mode));
+  if (this->m_mode == int(app::gen::SymmetryMode::NONE)) {
+    enabled(!enabled());
+  }
+  else {
+    enabled(true);
```

Comment:
Inside this `else` we can do the XOR of the `docPref` symmetry mode depending on the `m_mode` value. (because when m_mode is NONE, we don't even need the `docPref` variable).